hbase/stackable/patches/2.6.1/0005-Allow-overriding-ipc-bind-port-and-use-alternative-p.patch

```diff
@@ -0,0 +1,195 @@
+ import static org.apache.hadoop.hbase.namequeues.NamedQueueServiceChore.NAMED_QUEUE_CHORE_DURATION_DEFAULT;
+@@ -505,6 +506,10 @@ public class HRegionServer extends Thread
+    */
+   protected String useThisHostnameInstead;
+ 
++  protected int useThisPortInstead;
++
++  protected int useThisInfoPortInstead;
++
+   /**
+    * @deprecated since 2.4.0 and will be removed in 4.0.0. Use
+    *             {@link HRegionServer#UNSAFE_RS_HOSTNAME_DISABLE_MASTER_REVERSEDNS_KEY} instead.
+@@ -669,6 +674,8 @@ public class HRegionServer extends Thread
+       this.namedQueueRecorder = NamedQueueRecorder.getInstance(this.conf);
+       rpcServices = createRpcServices();
+       useThisHostnameInstead = getUseThisHostnameInstead(conf);
++      useThisPortInstead = getUseThisPortInstead(conf);
++      useThisInfoPortInstead = conf.getInt("hbase.info.port" , this.infoServer != null ? this.infoServer.getPort() : -1);
+ 
+       // if use-ip is enabled, we will use ip to expose Master/RS service for client,
+       // see HBASE-27304 for details.
+@@ -678,7 +685,7 @@ public class HRegionServer extends Thread
+         useIp ? rpcServices.isa.getAddress().getHostAddress() : rpcServices.isa.getHostName();
+       String hostName =
+         StringUtils.isBlank(useThisHostnameInstead) ? isaHostName : useThisHostnameInstead;
+-      serverName = ServerName.valueOf(hostName, this.rpcServices.isa.getPort(), this.startcode);
++      serverName = ServerName.valueOf(hostName, useThisPortInstead, this.startcode);
+ 
+       rpcControllerFactory = RpcControllerFactory.instantiate(this.conf);
+       rpcRetryingCallerFactory = RpcRetryingCallerFactory.instantiate(this.conf,
+@@ -715,7 +722,7 @@ public class HRegionServer extends Thread
+ 
+       // Some unit tests don't need a cluster, so no zookeeper at all
+       // Open connection to zookeeper and set primary watcher
+-      zooKeeper = new ZKWatcher(conf, getProcessName() + ":" + rpcServices.isa.getPort(), this,
++      zooKeeper = new ZKWatcher(conf, getProcessName() + ":" + useThisPortInstead, this,
+         canCreateBaseZNode());
+       // If no master in cluster, skip trying to track one or look for a cluster status.
+       if (!this.masterless) {
+@@ -776,6 +783,10 @@ public class HRegionServer extends Thread
+     }
+   }
+ 
++  protected int getUseThisPortInstead(Configuration conf) {
++    return conf.getInt(REGIONSERVER_PORT, this.rpcServices.isa.getPort());
++  }
++
+   private void setupSignalHandlers() {
+     if (!SystemUtils.IS_OS_WINDOWS) {
+       HBasePlatformDependent.handle("HUP", (number, name) -> {
+@@ -958,7 +969,7 @@ public class HRegionServer extends Thread
+       }
+       // Setup RPC client for master communication
+       this.rpcClient = RpcClientFactory.createClient(conf, clusterId,
+-        new InetSocketAddress(this.rpcServices.isa.getAddress(), 0),
++        new InetSocketAddress(0),
```

Comment:
I don't think the old behaviour here made sense _in general_, but I assume someone needed it at some point. Do we want to have a flag for reintroducing it?

(IMO, this probably makes less sense for a patch we carry for SDP, and more for being able to upstream it at some point.)

Reply:
Have added a flag that defaults to the new behaviour: bd0b75e

Reply:
de0c6c7